balloon.py

```diff
@@ -121,6 +121,25 @@ def balloon(
         time_cost (int): Number of rounds to mix.
         delta (int, optional): Number of random blocks to mix with. Defaults to 3.
 
+    Returns:
+        bytes: A series of bytes, the hash.
+    """
+    # Encode salt as bytes to be passed to _balloon()
+    return _balloon(password, salt.encode("utf-8"), space_cost, time_cost, delta)
+
+
+def _balloon(
```

Comment:
I've added an [internal](https://peps.python.org/pep-0008/#descriptive-naming-styles) _balloon() function to accept salt as **bytes** without mypy complaining about type mismatch.

This keeps the user-facing API unchanged (balloon() and balloon_m() are still defined as accepting salt as **str** only).